src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -791,9 +767,129 @@ private static synchronized Object getFifoNextSpanDataRaw(
       }
     }
+  }
 
-    // Return data as raw object (public method stub will convert to expected type)
-    return queueDataMap != null ? (Object) queueDataMap : (Object) queueDataList;
+  /**
+   * perform a rapid-catch up request and update tracking parameters. This request will attempt to
+   * find the start of data in an aggressive manner. If not data points are found, returns the end
+   * of the search timestamp. If data points are found, returns the timestamp of the first data
+   * point found. For proper behavior, Record data in UTC.
+   *
+   * @param startTimeTrackerMsLong the start time for historical data, in milliseconds since epoch
+   * @param includeTagGroupA if tag group A data should be included
+   * @param includeTagGroupB if tag group B data should be included
+   * @param includeTagGroupC if tag group C data should be included
+   * @param includeTagGroupD if tag group D data should be included
+   * @return timestamp in milliseconds since epoch
+   * @throws Exception for exceptions related to checking if data is exported in UTC format
+   * @throws IOException for errors closing exporter
+   */
+  private static long performRapidCatchUp(
+      long startTimeTrackerMsLong,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD)
+      throws IOException, Exception {
+
+    boolean stringHistorical = false;
+    boolean exportDataInUtc = true;
+    RapidCatchUpTracker catchUpResult =
+        RapidCatchUp.rapidCatchUpRequest(
+            startTimeTrackerMsLong,
+            includeTagGroupA,
+            includeTagGroupB,
+            includeTagGroupC,
+            includeTagGroupD,
+            stringHistorical,
+            exportDataInUtc);
+
+    if (catchUpResult.isHistoricalTrackingCaughtUp()) {
+      // If historical tracking is caught up, set lastReadDataPointsEmpty to false
+      lastReadDataPointsEmpty = false;
+    }
+
+    // Regardless of caught up status, update end time
+    return catchUpResult.getTrackingEndTimeMilliseconds();
+  }
+
+  /**
+   * Get the next span of historical data from the Historical Data Queue.
+   *
+   * @param ebdTagRequest The EBD request for the tag data
+   * @param ebdStringTagRequest The EBD request for the string tag data
+   * @param timeSpan Timespan object for round timestamps
+   * @param stringEnabled If string data is enabled
+   * @return A map of rounded timestamps to lists of data points
+   * @throws Exception For multiple possible exceptions reading historical data
+   */
+  private static Map getFifoNextSpanDataRawSpan(
+      String ebdTagRequest, String ebdStringTagRequest, SCTimeSpan timeSpan, boolean stringEnabled)
+      throws Exception {
+
+    // if enabled, read strings first
+    Map queueStringDataMap = null;
+    Map queueDataMap = null;
+
+    // it is important that the string tags are read first, and the duration between string and
+    // non-string tags is minimal.
```

Comment:
```suggestion
    // Non-string tags is minimal.
```

Reply:
This is not a new sentence or sentence fragment. It's the line-wrapped continuation of the comment. I will change to /**/ style comments to make that clear.